src/core/storage/fileio/s3_api.cpp

```diff
@@ -419,7 +426,9 @@ list_objects_response list_objects_impl(s3url parsed_url, std::string proxy,
     parsed_url.sdk_region = region;
   }
 
-  S3Client client(credentials, clientConfiguration);
+  S3Client client(credentials, clientConfiguration,
```

Comment:
This one doesn't seem to have your helpful inline comment. Those would be nice to add.

Reply:
Yeah, this duplicate part will be removed anyway.